### ISSUE TITLE
Media queries inside extraCss option can't be preserved

### ISSUE THREAD

**Issue by gakimball** [jonkemp/inline-css#26]
If media query CSS is passed into the `extraCss` option and `preserveMediaQueries` is `true`, the media query-specific CSS isn't preserved in the HTML output.

[Here's a small test case.](https://tonicdev.com/56aefb716c5d650c000b7384/56bb6d1b1c6cc30c00905fea)

The use case is running inline-css in the browser, where we're avoiding having to read or write from a filesystem. So instead of using `<link>` tag replacement, all of the CSS to be inlined is passed through `extraCss`.

Ideally the media query CSS in `extraCss` can be dropped into a `<style>` tag in the final output. I can think of a workaround, but it would be nice if the library could handle it.

Also an FYI, version 2 of [Foundation for Emails](http://foundation.zurb.com/emails) (formerly Ink) will be using this library to inline CSS :)

**Comment by jonkemp** [comment]
Thank you. I'm leaning toward categorizing this as a feature request rather than a bug. Are you ok with that?

**Comment by gakimball** [comment]
Yep, that's fine! I probably won't have time to implement it myself, but I have a workaround in mind either way.

**Comment by gakimball** [comment]
Updated initial comment to reference `applyLinkTags` along with `preserveMediaQueries`.

**Comment by martinrisseeuw** [comment]
@gakimball What is the workaround that you used to achieve this? I'm running into the same issue but can't figure out a workaround.

**Comment by gakimball** [comment]
> What is the workaround that you used to achieve this? I'm running into the same issue but can't figure out a workaround.

It's been a long time since I filed this issue, so I can't remember the specifics of what problem I was solving or how I solved it.

_However_, I have done some other HTML email work since then, and I ended up writing a library to handle all of this bundling stuff for me. You may find the source code helpful: <https://github.com/gakimball/scow>
